RSSReader-Maruyama/Modules/Login/LoginViewController.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  LoginViewController.swift
+//  RSSReader-Maruyama
+//
+//  Created by 丸山翔 on 2020/09/03.
+//  Copyright © 2020 丸山翔. All rights reserved.
+//
+
+import UIKit
+import AuthenticationServices
+import FBSDKCoreKit
+import FBSDKLoginKit
+import GoogleSignIn
+
+class LoginViewController: UIViewController {
+    
+    @IBOutlet weak var loginButtonStackView: UIStackView!
+    
+    private lazy var appleLoginButton: ASAuthorizationAppleIDButton = {
+        let appleLoginButton = ASAuthorizationAppleIDButton()
+        appleLoginButton.addTarget(self, action: #selector(didPushLoginButtonApple), for: .touchUpInside)
+        appleLoginButton.heightAnchor.constraint(equalToConstant: 45).isActive = true
+        
+        return appleLoginButton
+    }()
+    
+    private let SELECT_FEEDS_SEGUE_ID = "selectFeeds"
+    
+    fileprivate let userRepository = UserRepository()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        loginButtonStackView.addArrangedSubview(appleLoginButton)
+        GIDSignIn.sharedInstance()?.presentingViewController = self
+    }
+    
+    fileprivate func loginSuccessAction(userID: String) {
+        userRepository.save(user: User(id: userID))
+        // ログイン処理確認の画面遷移てテスト。実際はFeed選択画面へ
+        // self.performSegue(withIdentifier: SELECT_FEEDS_SEGUE_IDENTIFIER, sender: nil)
+        dismiss(animated: true, completion: nil)
+    }
+}
+
+// MARK: - Google Login Action
+
+extension LoginViewController {
+    @IBAction func didPushLoginButtonGoogle() {
+        GIDSignIn.sharedInstance()?.delegate = self
+        // ログインを実行
+        GIDSignIn.sharedInstance()?.signIn()
+    }
+}
```

Comment:
下記のようにDelegateも継承させてあげるとより良いかと思います。
また、処理を見ればすぐに分かるような箇所にはコメントは不要です。
良いコードというのはコメントが無くても処理が理解出来るものです。
コメントは仕様に伴った複雑な処理のみ記載するようにしましょう。

```suggestion
extension LoginViewController: GIDSignInDelegate  {
    @IBAction func didPushLoginButtonGoogle() {
        GIDSignIn.sharedInstance()?.delegate = self
        GIDSignIn.sharedInstance()?.signIn()
    }
    
    func sign(_ signIn: GIDSignIn!, didSignInFor user: GIDGoogleUser!, withError error: Error!) {
        if error == nil {
            loginSuccessAction(userID: user.userID)
        } else {
            print("error: \(error!.localizedDescription)")
        }
    }
}
```

Reply:
ありがとうございます。
GIDSignInDelegateで無いものを実装してしまうのは良くないと思っていました。
修正しました。